src/opera/util/input_validation.py

```diff
@@ -130,6 +131,133 @@ def validate_slc_s1_inputs(runconfig, logger, name):
+        'geometry_files
+    Verify that each file has a unique burst id in the file name, and that the burst idea is in the
+    set of of DISP_S1 cslc input burst ids.
+
+    Parameters
+    ----------
+    cslc_input_burst_ids : set
+        Set of the DISP_S1 input file burst_ids
+    ancillary_file_list : list
+        List from the runconfig file of the desired set of ancillary files to check
+    logger : PgeLogger
+        Logger passed by the calling PGE
+    name : str
+        PGE name
+
+    Raises
+    ------
+    RuntimeError
+        If there are ancillary files of the same type with the same burst_id.
+        If the set of ancillary file burst_ids do not match the set of cslc input file burst_ids.
+
+    """
+    ancillary_burst_ids = get_burst_id_set(ancillary_file_list, logger, name)
+    # Test that there is a unique burst_id for each burst
+    if len(ancillary_burst_ids) != len(ancillary_file_list):
+        msg = f'Burst ids in {ancillary_file_list} are not unique.'
+        logger.critical(name, ErrorCode.INVALID_INPUT, msg)
+    # Verify that the sets of burst_ids are equal
+    if ancillary_burst_ids != cslc_input_burst_ids:
+        msg = f'input_burst_ids: {cslc_input_burst_ids} do not match ancillary_burst_ids: ' \
+              f'{ancillary_burst_ids} for {ancillary_file_list}'
+        logger.critical(name, ErrorCode.INVALID_INPUT, msg)
+
+
+def get_cslc_input_burst_id_set(cslc_input_file_list, logger, name):
+    """
+    Compile the set of burst ids from the cslc input files
+    There may be uncompressed files only, or a mixture of uncompressed
+    and compressed files.  If mixed the uncompressed set of burst ids
+    must match the set of compressed burst ids.
+
+    Parameters
+    ----------
+    cslc_input_file_list : list
+        List of all cslc input files pulled from the runconfig file
+    logger : PgeLogger
+        Logger passed by the calling PGE
+    name : str
+        PGE name
+
+    Returns
+    -------
+    burst_id_set :set
+
+    Raises
+    ------
+    RuntimeError
+        If the compressed burst_ids set does not match the uncompressed burst_id set.
+
+    """
+    compressed_input_file_list = []
+    single_input_file_list = []
+
+    for i in cslc_input_file_list:
+        if 'compressed' in i:
+            compressed_input_file_list.append(i)
+        else:
+            single_input_file_list.append(i)
+
+    compressed_ids = get_burst_id_set(compressed_input_file_list, logger, name)
+    single_file_burst_id_set = get_burst_id_set(single_input_file_list, logger, name)
+
+    # Case 1:  uncompressed files only in cslc inputs
+    if len(compressed_ids) == 0:
+        return single_file_burst_id_set
+    # Case 2: uncompressed files and compressed files in cslc inputs with non-matching burst ids
+    elif single_file_burst_id_set != compressed_ids:
+        msg = f"single_file_burst_id_set: {single_file_burst_id_set} does not match compressed_file_burst_id_set: " \
```

Comment:
See my message above about how to format this error message to only print out the differences between the two sets